test/src/main/java/org/apache/accumulo/test/ample/TestAmple.java

```diff
@@ -0,0 +1,212 @@
+    }
+
+    /**
+     * Create default metadata for a Table
+     *
+     * TODO: Add a way to pass in options for config
+     *
+     * @param tableId The id of the table to create metadata for
+     */
+    public void createMetadata(TableId tableId) {
+      try (var tabletsMutator = mutateTablets()) {
+        var extent = new KeyExtent(tableId, null, null);
+        var tabletMutator = tabletsMutator.mutateTablet(extent);
+        String dirName = ServerColumnFamily.DEFAULT_TABLET_DIR_NAME;
+        tabletMutator.putPrevEndRow(extent.prevEndRow());
+        tabletMutator.putDirName(dirName);
+        tabletMutator.putTime(new MetadataTime(0, TimeType.MILLIS));
+        tabletMutator.putTabletAvailability(TabletAvailability.HOSTED);
+        tabletMutator.mutate();
+      } catch (Exception e) {
+        throw new IllegalStateException(e);
+      }
+    }
+
+    /**
+     * Create metadata for a table by copying existing metadata for the table from the metadata
+     * table in an existing Accumulo instance
+     *
+     * TODO: Add config parents (such as a way to include/exclude what is copied, etc)
+     *
+     * @param client The client to scan the existing accumulo metadata table
+     * @param tableId The id of the table to create metadata for
+     * @throws Exception thrown for any error on metadata creation
+     */
+    public void createMetadataFromExisting(AccumuloClient client, TableId tableId)
+        throws Exception {
+      createMetadataFromExisting(client, tableId, Set.of());
+    }
+
+    public void createMetadataFromExisting(AccumuloClient client, TableId tableId,
+        Set<ColumnFQ> excludedColumnFq) throws Exception {
+      try (Scanner scanner =
+          client.createScanner(AccumuloTable.METADATA.tableName(), Authorizations.EMPTY)) {
+        scanner.setRange(TabletsSection.getRange(tableId));
+        IteratorSetting iterSetting = new IteratorSetting(100, WholeRowIterator.class);
+        scanner.addScanIterator(iterSetting);
+
+        try (BatchWriter bw = client.createBatchWriter(getMetadataTableName())) {
+          for (Entry<Key,Value> entry : scanner) {
+            final SortedMap<Key,Value> decodedRow =
+                WholeRowIterator.decodeRow(entry.getKey(), entry.getValue());
+            Text row = decodedRow.firstKey().getRow();
+            Mutation m = new Mutation(row);
+
+            decodedRow.entrySet().stream()
+                .filter(e -> !excludedColumnFq.contains(new ColumnFQ(e.getKey()))).forEach(e -> {
+                  m.put(e.getKey().getColumnFamily(), e.getKey().getColumnQualifier(),
+                      e.getKey().getColumnVisibilityParsed(), e.getKey().getTimestamp(),
+                      e.getValue());
+                });
+            bw.addMutation(m);
+          }
+        }
+      }
+    }
+
+    public ConditionalTabletsMutator
+        conditionallyMutateTablets(ConditionalWriterInterceptor interceptor) {
```

Comment:
Having this interceptor will be great for testing.  Need some way to use it when code only has a handle to the Ample interface.   Wonder if we could pass something like `Supplier< ConditionalWriterInterceptor> interceptorFactory` to the  TestAmple constructor.   Then could do the following in TestAmple

```java
    @Override
    public ConditionalTabletsMutator conditionallyMutateTablets() {
      return conditionallyMutateTablets(interceptorFactory.get());
    }
```

Reply:
Added in [e5feb49](https://github.com/apache/accumulo/pull/4415/commits/e5feb4960ccc06430330650ec057dcd8aab5adad)